addons/mass_mailing/models/res_company.py

```diff
@@ -7,11 +7,13 @@
 class ResCompany(models.Model):
     _inherit = "res.company"
 
-    def _get_social_media_links(self):
-        self.ensure_one()
-        return {
-            'social_facebook': self.social_facebook,
-            'social_linkedin': self.social_linkedin,
-            'social_twitter': self.social_twitter,
-            'social_instagram': self.social_instagram
-        }
+    def try_set_social_media_links(self, values):
```

Comment:
Why not calling write directly then ? What are you really trying to do ? Moreover using "try" implies a try/catch which is not the case here.